src/charm.py

```diff
@@ -345,6 +358,9 @@ def _on_collect_status(self, event: CollectStatusEvent) -> None:  # noqa: C901
                 BlockedStatus("Missing certificate transfer integration with oauth provider")
             )
 
+        if not smtp_integration_exists(self):
+            event.add_status(BlockedStatus(f"Missing integration {SMTP_INTEGRATION_NAME}"))
```

Comment:
I consider the `smtp` integration a must because the underlying admin service is not considered fully functional without a proper mail service setup.